atomium/includes/common.inc

```diff
@@ -667,33 +667,125 @@ function _atomium_array_flatten(array $array, array &$storage, $parentKey = '')
  */
 function atomium_get_settings($setting_keys, $base_themes = TRUE, array $value_callbacks = array('trim')) {
   $theme_key = $GLOBALS['theme_key'];
+  $settings_flat_filtered = _atomium_get_settings_flat_filtered($theme_key, $base_themes, $value_callbacks);
 
-  $settings = array();
+  return isset($settings_flat_filtered[$setting_keys])
+    ? $settings_flat_filtered[$setting_keys]
+    : [];
+}
 
-  if ($theme_settings = atomium_get_theme_info($theme_key, 'settings', $base_themes)) {
-    _atomium_array_flatten(
-      $theme_settings,
-      $settings
-    );
+/**
+ * @param string $theme_key
+ * @param bool $base_themes
+ * @param callable[] $value_callbacks
+ *
+ * @return mixed
+ */
+function _atomium_get_settings_flat_filtered($theme_key, $base_themes = TRUE, $value_callbacks = ['trim']) {
+  static $settings_flat_filtered_by_cid = [];
```

Comment:
Why not use drupal_static?

Reply:
Please could you address your remarks in #201 ?